.github/workflows/pr_release_artifacts.yml

```diff
@@ -0,0 +1,137 @@
+name: pr release artifacts
+
+on:
+  schedule:
+    - cron: '0 15 * * 4'
+  workflow_dispatch:
+    inputs:
+      cleanup:
+        description: Run cleanup instead of creating artifacts
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  create-release-artifacts:
+    name: Create release artifacts for PR
+    if: github.event_name == 'workflow_dispatch' && github.event.inputs.cleanup == 'false'
+    runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write
+      issues: write
+      contents: read
+      packages: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      - uses: ./.github/actions/setup_nix
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          cachixToken: ${{ secrets.CACHIX_AUTH_TOKEN }}
+      - name: Log in to ghcr.io Container registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3.4.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get pre-release version
+        id: get-version
+        run: |
+          echo "version=$(cat version.txt)" | tee -a "$GITHUB_OUTPUT"
+      - name: Configure git
+        run: |
+          git config --global user.name "edgelessci"
+          git config --global user.email "edgelessci@users.noreply.github.com"
+      - name: Create release artifacts
+        id: create-artifacts
+        uses: ./.github/actions/release_artifacts
+        with:
+          version: ${{ steps.get-version.outputs.version }}
+          container_registry: ghcr.io/edgelesssys
+          s3-bucket-path: "pr-artifacts"
+      - name: Get PR number
+        id: get-pr-number
+        env:
+          COMMIT_SHA: ${{ github.sha }}
+          REPO: ${{ github.repository }}
+        run: |
+          prs=$(
+            curl -fsSL \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+              "https://api.github.com/repos/${REPO}/commits/${COMMIT_SHA}/pulls"
+          )
+          pr=$(echo "$prs" | jq -r '.[0]?.number // ""')
+          echo "pr_number=$pr" | tee -a "$GITHUB_OUTPUT"
+      - name: Create PR comment with artifact links
+        if: ${{ steps.get-pr-number.outputs.pr_number != '' }}
+        uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v4.0.0
+        with:
+          issue-number: ${{ steps.get-pr-number.outputs.pr_number }}
+          body: |
+            ### Pre-release artifacts on ${{ github.sha }}
+
+            The pre-release artifacts for this commit are available at the following link:
+
+            ${{ steps.create-artifacts.outputs.s3-bucket-url }}
```

Comment:
Is this URL supposed to be publicly accessible? It's not for me.

Reply:
It is, but you can't list. You need to append the name of an existing artifact to the path.

Reply:
Guess I'm just not the target audience for the links.